lua/auto-save/init.lua

```diff
@@ -148,32 +148,18 @@ function M.on()
     desc = "Cancel a pending save timer for a buffer",
   })
 
-  api.nvim_create_autocmd({ "VimEnter", "ColorScheme", "UIEnter" }, {
-    callback = function()
-      vim.schedule(function()
-        if cnf.opts.execution_message.dim > 0 then
-          MSG_AREA = colors.get_hl("MsgArea")
-          if MSG_AREA.foreground ~= nil then
-            MSG_AREA.background = (MSG_AREA.background or colors.get_hl("Normal")["background"])
-            local foreground = (
-              o.background == "dark"
-                and colors.darken(
-                  (MSG_AREA.background or BLACK),
-                  cnf.opts.execution_message.dim,
-                  MSG_AREA.foreground or BLACK
-                )
-              or colors.lighten(
-                (MSG_AREA.background or WHITE),
-                cnf.opts.execution_message.dim,
-                MSG_AREA.foreground or WHITE
-              )
-            )
-
-            colors.highlight("AutoSaveText", { fg = foreground })
-            AUTO_SAVE_COLOR = "AutoSaveText"
-          end
-        end
+  local function dim()
+    if cnf.opts.execution_message.enabled then
+      schedule(function()
+        colors.apply_colors(cnf.opts.execution_message.dim)
       end)
+    end
+  end
+
+  dim()
```

Comment:
I think this should be called `setup_dimming` or `config_dimming` or something like that since it's not actually dimming anything at call time.